### PR TITLE
Spheroid dropship movement

### PR DESCRIPTION
**Pull request by NickAragua**
Fixes several "bugs" with airborne spheroid dropships in low atmosphere and on ground maps, such as them being unable to move at all on the former and zigzagging like a crazed teenager on the ground maps. Also addresses a few issues with firing arcs and enables the bot to use flying spheroids properly.

Detailed list:
- Player can now move spheroid dropships on low-atmo maps using TW rules (1 hex)
- Spheroid dropship movement on ground map is restricted to 8 hexes per TW and must follow jump-jet-like pattern (not explicitly in the rules, but...)
- Spheroid dropships can properly fire at higher airborne targets with nose weapons, and lower airborne targets with aft weapons.
- Bot can use spheroid dropships according to above rules
- Incidentally, bot infantry can now rotate after movement
- Fixed several issues where the bot was using the wrong FireControl class to evaluate damage potential
- Player must now manually "hover" their spheroid dropship if not changing altitude explicitly.

Addresses issue #1303

### CODE REVIEW

File: megamek/src/megamek/client/bot/princess/PathEnumerator.java [hunk 1]
```diff
@@ -250,13 +251,18 @@ public boolean shouldStay(MovePath movePath) {
...
                 apf.run(new MovePath(game, mover));
                 paths.addAll(apf.getAllComputedPathsUncategorized());
-            // this handles the case of the mover being an infantry unit of some kind
-            } else if(mover.isAero() && game.getBoard().inAtmosphere()) {
+            // this handles the case of the mover being a winged aerospace unit on a low-atmo map
+            } else if(mover.isAero() && game.getBoard().inAtmosphere() && !Compute.useSpheroidAtmosphere(game, mover)) {
                 AeroLowAltitudePathFinder apf = AeroLowAltitudePathFinder.getInstance(getGame());
                 apf.run(new MovePath(game, mover));
                 paths.addAll(apf.getAllComputedPathsUncategorized());
+            // this handles the case of the mover acting like a spheroid aerospace unit in an atomsphere
```
Comment (sixlettervariables) [inline megamek/src/megamek/client/bot/princess/PathEnumerator.java]:
```suggestion
            // this handles the case of the mover acting like a spheroid aerospace unit in an atmosphere
```

File: megamek/src/megamek/common/MoveStep.java [hunk 2]
```diff
@@ -658,16 +660,15 @@ private void compileMove(final IGame game, final Entity entity,
         // if this is a flying aero, then there is no MP cost for moving
         if ((prev.getAltitude() > 0) || game.getBoard().inSpace()) {
             setMp(0);
-            // if this a spheroid in atmosphere then the cost is always two
+            // if this a spheroid in atmosphere then the cost is always one
```
Comment (sixlettervariables) [inline megamek/src/megamek/common/MoveStep.java]:
```suggestion
            // if this is a spheroid in atmosphere then the cost is always one
```

File: megamek/src/megamek/common/MoveStep.java [hunk 3]
```diff
@@ -756,6 +757,7 @@ protected void compile(final IGame game, final Entity entity, MoveStep prev) {
         boolean isFieldArtillery = (entity instanceof Infantry)
                 && ((Infantry) entity).hasActiveFieldArtillery();
         copy(game, prev);
+        int alpha = 1;
```
Comment (sixlettervariables) [inline megamek/src/megamek/common/MoveStep.java]: is this being used?

File: megamek/src/megamek/common/pathfinder/InfantryPathFinder.java [hunk 4]
```diff
@@ -91,19 +69,19 @@ public void run(MovePath startingEdge) {
             jumpEdge.addStep(MoveStepType.START_JUMP);
             infantryPaths.addAll(generateChildren(jumpEdge));
             
+            // now that we've got all our possible destinations, make sure to try every possible rotation,
+            // since facing matters for field guns and if using the "dig in" and "vehicle cover" tacops rules.
+            for(MovePath path : infantryPaths) {
+                infantryPaths.addAll(AeroPathUtil.generateValidRotations(path));
```
Comment (sixlettervariables) [inline megamek/src/megamek/common/pathfinder/InfantryPathFinder.java]: Curiosity's sake: why the `AeroPathUtil` here?

File: megamek/src/megamek/common/pathfinder/SpheroidPathFinder.java [hunk 5]
```diff
@@ -0,0 +1,212 @@
+package megamek.common.pathfinder;
```
Comment (sixlettervariables) [inline megamek/src/megamek/common/pathfinder/SpheroidPathFinder.java]: Copyright header

File: megamek/src/megamek/common/pathfinder/InfantryPathFinder.java (hunk 4 above)
Comment (NickAragua) [reply megamek/src/megamek/common/pathfinder/InfantryPathFinder.java]: That's where the generateValidRotations() method lives.

File: megamek/src/megamek/client/bot/princess/AeroPathUtil.java [hunk 6]
```diff
@@ -190,4 +219,25 @@ public static int calculateMaxSafeThrust(IAero aero) {
...
+
+    /**
+     * Given a move path, generates all possible rotations from it,
+     * without any regard to legality. Mostly because it's intended for spheroid 
+     * dropships in atmosphere, which can rotate as much as they want.
+     */
+    public static List<MovePath> generateValidRotations(MovePath path) {
+        List<MovePath> childPaths = new ArrayList<>();
+        
+        for(int x = 1; x < TURNS.size(); x++) {
```
[the remark this answers is not in the document]
Comment (NickAragua) [reply megamek/src/megamek/client/bot/princess/AeroPathUtil.java]: Yeah, in this case we don't need the "no rotation" rotation.

**Pull request merged**